### PR TITLE
Use QueryProducts to queue 'currencyCode' state change on account level 'Add Payment Method' page

### PR DESCRIPTION
**Pull request by jjchrisdiehl**
As noted in the project thread, the original implementation of the `<CBLogo />` within the `PaymentMethodSelector` component relied on a currency code value in order to conditionally show the CBLogo only to customers who have their account currency set to `EUR`. 

Originally, this value was taken from the state value `currencyCode` which initially worked for both site level and account level versions of the Add Payment Method pages...sort of.

Since `currencyCode` is populated by only three actions, it relies on specific network requests to `plans` and `products` to update it:

https://github.com/Automattic/wp-calypso/blob/28740d3224a8fc659adc6d056ad6b424826bbe7d/client/state/currency-code/reducer.js#L13-L19

This is fine for the site level 'Add Payment Method' page, and even for the account level if you navigate to it from a section that already triggered these actions and set the `currencyCode` state. 

But, if you were to reload the account level version, the `currencyCode` state is never populated since `me/purchases/add-payment-method` never makes these network requests and therefore doesn't call the actions.

This isn't the case for the site level version, since the plans endpoint is called when Calypso loads the sidebar (I think).

Related to https://github.com/Automattic/payments-shilling/issues/3092

## Proposed Changes

In `AccountLevelAddNewPaymentMethodWrapper` we can run `QueryProducts` to populate the `currencyCode` state in redux. 

I also pass an array of a couple dotcom product slugs in order to speed up the request, instead of calling for every single product available. This creates a noticable impact on how quickly the CBLogo is rendered.

## Testing Instructions
* Set account currency to `EUR`
* Go directly to http://calypso.localhost:3000/me/purchases/add-payment-method (don't navigate via /me/purchases > Add Payment Method)
* Ensure that the CBLogo loads alongside Visa, Mastercard, and AMEX. 
* Refresh the page, and make sure it still loads

image

### CODE REVIEW

**Review by sirbrillig — APPROVED** [review]
It's really weird that the user's currency in Redux comes from the products endpoint 🙃. Nice discovery! It would be nice to create a new custom hook to fetch the currency directly and skip Redux entirely for this purpose, but in the meantime this is a good solution. I'm glad you added a comment to explain what's happening.

This works as described!

Before             |  After
:-------------------------:|:-------------------------:
before-cb | after-cb

File: client/me/purchases/add-new-payment-method/index.tsx [hunk 1]
```diff
@@ -87,9 +87,19 @@ function AddNewPaymentMethod() {
...
 			<RazorpayHookProvider fetchRazorpayConfiguration={ getRazorpayConfiguration }>
+				{
+					// QueryProducts added to ensure currency-code state gets populated for usage of getCurrentUserCurrencyCode
+					// Returning a couple of standard products to speed up the render time and ensure redundancy
+					// We only need one to get the currency code into state
+				 }
+				<QueryProducts
+					productSlugList={ [ 'value_bundle', 'personal-bundle', 'business-bundle' ] }
+				/>
+				;
```
Comment (michaeldcain) [inline client/me/purchases/add-new-payment-method/index.tsx]: Is this a typo?

**Review by michaeldcain — DISMISSED** [review]
There's a typo that needs to be fixed.

File: client/me/purchases/add-new-payment-method/index.tsx (hunk 1 above)
Comment (jjchrisdiehl) [reply client/me/purchases/add-new-payment-method/index.tsx]: Gremlins, fixed.

**Comment by jjchrisdiehl** [comment]
> It's really weird that the user's currency in Redux comes from the products endpoint 🙃. Nice discovery! It would be nice to create a new custom hook to fetch the currency directly and skip Redux entirely for this purpose, but in the meantime this is a good solution. I'm glad you added a comment to explain what's happening.

Yeah, it is sort of weird since we aren't going to use the products at all which feels like some sort of anti-pattern. I noticed it while looking for ways to trigger the actions for currencyCode. I figured I could try to pass it from the server but came across this method a [number of times](https://github.com/search?q=repo%3AAutomattic%2Fwp-calypso%20%2F**%20QueryProducts%20added&type=code) in the existing code and figured it handled the job well enough with no obvious degradation in performance. 

I've [opened an issue](https://github.com/Automattic/payments-shilling/issues/3098) to add a useUserCurrency hook in the near future.

**Pull request merged**